### DISCUSSION TITLE
Can Plausible be used with Apache reverse proxy?

### DISCUSSION THREAD

**Discussion by TheNomad11** [plausible/analytics#765] [Self-Hosted Support] [answered]
Say, I already have Apache running as a service om my VPS. Can I still use Plausible (with docker)? In the documentation I only see reverse proxy examples with Caddy and Nginx. Thanks a lot!

**Comment by Vigasaurus** [comment]
Plausible can indeed run through Apache as well, examples are just only provided for Nginx and caddy. See: https://www.digitalocean.com/community/tutorials/how-to-use-apache-http-server-as-reverse-proxy-using-mod_proxy-extension for how to do so (not specific, but basic instructions apply).

**Comment by ukutaht** [comment]
Sure, go ahead! The only reason we don't have docs for Apache is that I've only ever used Nginx and Caddy myself. If you do find the right config for Apache, I'd love to have the instructions contributed back to our documentation.

**Comment by Philipinho** [comment, marked as answer]
### How to set up Apache reverse proxy for Plausible Analytics.
This is for Ubuntu but should easily be adapted for any other OS.

**Note:** replace plausible.example.com with your desired domain.

1. **Enable the necessary modules**
`sudo a2enmod proxy proxy_http proxy_ajp remoteip headers`

2. **Restart apache**
`sudo systemctl restart apache2`

3. **Create a virtual host block for your domain**
`sudo vi /etc/apache2/sites-available/plausible.example.com.conf`

4. **Add the following configurations to the newly created virtual host file.**
    ```
    <VirtualHost *:80>
    
     ServerAdmin admin@example.com
     ServerName plausible.example.com
    
     ProxyPreserveHost On
     ProxyPass / http://localhost:8000/
     ProxyPassReverse / http://localhost:8000/
    
     SetEnvIf X-Forwarded-For "^.*\..*\..*\..*" forwarded
     LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\"" combined
     LogFormat "%{X-Forwarded-For}i %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\"" forwarded
    
     ErrorLog ${APACHE_LOG_DIR}/error.log
     CustomLog ${APACHE_LOG_DIR}/access.log combined env=!forwarded
     CustomLog ${APACHE_LOG_DIR}/access.log forwarded env=forwarded
	    
    </VirtualHost>
    ```

5.  **Enable the virtual host block** 
`sudo a2ensite plausible.example.com.conf`

6. **Restart apache**
`sudo systemctl restart apache2`

That is all.

**Reply by TheNomad11** [reply]
Thanks a lot. Will try it later!

**Reply by ukutaht** [reply]
Awesome, thanks! No pressure but it would be very helpful to add it to our hosting repo for future reference: https://github.com/plausible/hosting/tree/master/reverse-proxy
